### PR TITLE
fix: abandon updates if timestamp.json isn't new

### LINKED ISSUE #321: TUF specification has a new version - v1.0.30
Hey, it seems there's a newer version of the TUF specification - [v1.0.30](https://github.com/theupdateframework/specification/blob/v1.0.30/tuf-spec.md)

The version which [theupdateframework/go-tuf](https://github.com/theupdateframework/go-tuf) states it supports is - [v1.0.29](https://github.com/theupdateframework/specification/blob/v1.0.29/tuf-spec.md)

The following is a comparison of what changed between the two versions - [Compare v1.0.29 to v1.0.30](https://github.com/theupdateframework/specification/compare/v1.0.29...v1.0.30)

Please review the newer version and address the changes.

### PR DESCRIPTION
**Pull request by znewman01**
Adds a new test for this case: if a client sees a new `timestamp.json` file with the same version as its current `timestamp.json` file, it should do nothing (no update, but also no error).

A few other tests were implicitly relying on the fact that the client did a full update each time, so they've been updated to commit a new timestamp.

This updates go-tuf for TUF specification v1.0.30 (fixes #321). The only substantive change was
[theupdateframework/specification#209][tuf-spec-209], which clarifies the intended behavior for updating metadata files.

Updates for other roles were already in compliance:

- Root metadata: https://github.com/theupdateframework/go-tuf/blob/13eff30efd6c61f165e1bf06e8c0e72f5a0e5703/client/client.go#L258
- Timestamp, checking snapshot version: https://github.com/theupdateframework/go-tuf/blob/13eff30efd6c61f165e1bf06e8c0e72f5a0e5703/client/client.go#L751
- Snapshot, must match version from timestamp: https://github.com/theupdateframework/go-tuf/blob/13eff30efd6c61f165e1bf06e8c0e72f5a0e5703/client/client.go#L667
- Snapshot, no rollback of targets: https://github.com/theupdateframework/go-tuf/blob/13eff30efd6c61f165e1bf06e8c0e72f5a0e5703/client/client.go#L685
- Targets: https://github.com/theupdateframework/go-tuf/blob/13eff30efd6c61f165e1bf06e8c0e72f5a0e5703/client/client.go#L643

[tuf-spec-209]: (https://github.com/theupdateframework/specification/pull/209).

Signed-off-by: Zachary Newman <z@znewman.net>

---------------------------------------------------------

Please fill in the fields below to submit a pull request.  The more information that is provided, the better.

Fixes #369
Release Notes: go-tuf clients now only complete an update if the timestamp.json file is new, rather than completing the same update over and over.

**Types of changes**:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

**Description of the changes being introduced by the pull request**: See above

**Please verify and check that the pull request fulfills the following requirements**:

- [x] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature: N/A

### CODE REVIEW

File: client/client_test.go [hunk 1]
```diff
@@ -964,6 +966,43 @@ func (s *ClientSuite) TestUpdateReplayAttack(c *C) {
...
+
+	// generate a new, different timestamp with the *same version*
+	s.remote.meta["timestamp.json"] = oldTimestamp
+	c.Assert(s.repo.Timestamp(), IsNil)
+	c.Assert(client.timestampVer, Equals, newVersion) // double-check: same version?
+	s.syncRemote(c)
+
+	oldMeta, err := client.local.GetMeta()
+	c.Assert(err, IsNil)
+	fmt.Println("here we go")
```
Comment (joshuagl) [inline client/client_test.go:996]: Nit: we should probably remove this `Println()` call?

**Review by joshuagl — DISMISSED** [review]
Great stuff, thanks.

File: client/client_test.go (hunk 1 above)
Comment (znewman01) [reply client/client_test.go:996]: ha, can you tell I was debugging?

Comment (joshuagl) [reply client/client_test.go:996]: had to check I hadn't written it myself 😆

File: client/client_test.go [hunk 2]
```diff
@@ -964,6 +966,43 @@ func (s *ClientSuite) TestUpdateReplayAttack(c *C) {
...
+	// check update returns ErrLowVersion
```
[the remark this answers is not in the document]
Comment (joshuagl) [reply client/client_test.go:997]: This is still here, but we don't actually check what `err` is. Is this comment incorrect or is my meagre grasp of Go failing me?

[the remark this answers is not in the document]
Comment (joshuagl) [reply client/client_test.go:997]: To my reading, what we're actually checking is that update does not return any error

[the remark this answers is not in the document]
Comment (znewman01) [reply client/client_test.go]: I `git push`ed and closed the terminal without checking that it went through ☹️ 

It's removed now. Your interpretation is right: we want it *not* to return any error.

(There was discussion in the spec about having clients check for the same version with different content and returning an error, but the spec as it reads today tells us to ignore it.)

**Review by mnm678 — APPROVED** [review]

**Review by joshuagl — APPROVED** [review]

**Pull request merged**